src/api/fleets.spec.ts

```diff
@@ -0,0 +1,1641 @@
+    DoTest(options.testName + ' - Default test cases - emptyString',
+    options.parameterToTest, options.pathToParameterToTest, '', options.emptyStringExpectedResponse);
+    DoTest(options.testName + ' - Default test cases - null',
+    options.parameterToTest, options.pathToParameterToTest, null, options.nullExpectedResponse);
+    DoTest(options.testName + ' - Default test cases - true',
+    options.parameterToTest, options.pathToParameterToTest, true, options.trueExpectedResponse);
+    DoTest(options.testName + ' - Default test cases - false',
+    options.parameterToTest, options.pathToParameterToTest, false, options.falseExpectedResponse);
+    DoTest(options.testName + ' - Default test cases - maxSafeInteger',
+    options.parameterToTest, options.pathToParameterToTest, Number.MAX_SAFE_INTEGER, options.maxSafeIntegerExpectedResponse);
+    DoTest(options.testName + ' - Default test cases - minSafeInteger',
+    options.parameterToTest, options.pathToParameterToTest, Number.MIN_SAFE_INTEGER, options.minSafeIntegerExpectedResponse);
+    DoTest(options.testName + ' - Default test cases - undefined',
+    options.parameterToTest, options.pathToParameterToTest, undefined, options.undefinedExpectedResponse);
+
+    // The custom test cases which are specifically defined for the current field.
+    if (options.customTestCases) {
+      options.customTestCases.forEach(customTestCase => {
+        DoTest(options.testName + ' - Custom test cases - ' + customTestCase.customTestCaseValue, options.parameterToTest,
+        options.pathToParameterToTest, customTestCase.customTestCaseValue, customTestCase.customTestCaseExpectedResponse);
+      });
+    }
+  }
+
+  // Dynamically set property of nested object
+  // I got it from here: https://stackoverflow.com/questions/18936915/dynamically-set-property-of-nested-object
+  function setProperty(obj, path, value) {
+    let schema = obj;
+    const pList = path.split('.');
+    const len = pList.length;
+    for (let i = 0; i < len - 1; i++) {
+        const elem = pList[i];
+        if ( !schema[elem] ) {
+          schema[elem] = {}
+        }
+        schema = schema[elem];
+    }
+    schema[pList[len - 1]] = value;
+  }
+
+  // Execute a test, this method is executed for each edge case that's defined for a field.
+  function DoTest(
+  testName: string,
+  parameterToTest: ParameterToTest,
+  pathToPropertyInParameterToTest: string,
+  testValue: any,
+  expectedResponse: ExpectedResponse) {
+    if (expectedResponse != ExpectedResponse.DONT_EXECUTE_THIS_TEST) {
+      // Clone the base request data.
+      const options: FleetRequestOptionsImplementation = JSON.parse(JSON.stringify(optionsBase));
+      const stores: StoreImplementation[] = JSON.parse(JSON.stringify(storesBase));
+      const transports: TransportImplementation[] = JSON.parse(JSON.stringify(transportsBase));
+      const orders: OrderImplementation[] = JSON.parse(JSON.stringify(ordersBase));
```

Comment:
can you explain why this is necessary: `JSON.parse(JSON.stringify(ordersBase));`

Reply:
This is to make a copy of the object, this copy will then be changed for the test. The changes will not reflect in the base object as well so it can be copied again for another test.